clang/lib/Sema/SemaDecl.cpp

```diff
@@ -19474,11 +19474,12 @@ EnumConstantDecl *Sema::CheckEnumConstant(EnumDecl *Enum,
           //   representable as an int.
 
           // Complain if the value is not representable in an int.
-          if (!isRepresentableIntegerValue(Context, EnumVal, Context.IntTy))
-            Diag(IdLoc, diag::ext_enum_value_not_int)
-              << toString(EnumVal, 10) << Val->getSourceRange()
-              << (EnumVal.isUnsigned() || EnumVal.isNonNegative());
-          else if (!Context.hasSameType(Val->getType(), Context.IntTy)) {
+          if (!isRepresentableIntegerValue(Context, EnumVal, Context.IntTy)) {
+            if (!getLangOpts().C23)
+              Diag(IdLoc, diag::ext_enum_value_not_int)
```

Comment:
There should be a `-Wpre-c23-compat` warning for this, e.g. `warn_c17_compat_enum_value_not_int` `InGroup<CPre23Compat>, DefaultIgnore`

Reply:
Added.